src/test/ui/consts/issue-70114.rs

```diff
@@ -1,12 +1,34 @@
-// build-pass
-
+// run-pass
 // Test a ZST enum whose dicriminant is ~0i128. This caused an ICE when casting to a i32.
 
 enum Nums {
     NegOne = -1,
 }
 
+#[inline(never)]
+fn identity<T>(t: T) -> T { t }
+
+fn test_as_arg(n: Nums) {
+    assert_eq!(-1i8, n as i8);
+    assert_eq!(-1i16, n as i16);
+    assert_eq!(-1i32, n as i32);
+    assert_eq!(-1i64, n as i64);
+    assert_eq!(-1i128, n as i128);
+}
+
 fn main() {
     let kind = Nums::NegOne;
-    let _v = kind as i32;
+    assert_eq!(-1i8, kind as i8);
+    assert_eq!(-1i16, kind as i16);
+    assert_eq!(-1i32, kind as i32);
+    assert_eq!(-1i64, kind as i64);
+    assert_eq!(-1i128, kind as i128);
+
+    assert_eq!(-1i8, identity(kind) as i8);
+    assert_eq!(-1i16, identity(kind) as i16);
+    assert_eq!(-1i32, identity(kind) as i32);
+    assert_eq!(-1i64, identity(kind) as i64);
+    assert_eq!(-1i128, identity(kind) as i128);
+
+    test_as_args(Nums::NegOne);
```

Comment:
Could you also have one that explicitly uses consts?
```rust
const NEG_ONE_I8: i8 = Nums::NegOne as i8;

assert_eq!(-1, NEG_ONE_I8);
```

Reply:
[Yes, absolutely](https://github.com/rust-lang/rust/pull/70126/commits/61ab908df652d482d8ce4b5530229ffa293ae766). Sorry, I missed the earlier mention of this amongst the other comments.